packages/@react-spectrum/list/src/ListView.tsx

```diff
@@ -82,17 +82,18 @@ function ListView<T extends object>(props: ListViewProps<T>, ref: DOMRef<HTMLDiv
     columnCount: 1,
     items: [...collection].map(item => ({
       type: 'item',
+      key: item.key,
       childNodes: [{
         ...item,
         index: 0,
-        type: 'cell'
+        type: 'cell',
+        key: `cell-${item.key}`
```

Comment:
Should we perhaps make the row the item, and synthesize a cell object here? I had an example that was working in the dnd stories: https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/dnd/stories/DraggableCollection.tsx

Might have to make some changes to ListViewItem as well. Looks like both `useGridRow` and `useGridCell` are passed the same item node, which means they get the same key and will fight for focus.

Reply:
I've pushed up the changes, but I'm running into a snag with https://github.com/adobe/react-spectrum/blob/66d8eb4f4da3c073f4184f70ff6fcad47a1e8238/packages/%40react-aria/virtualizer/src/Virtualizer.tsx#L164 moving focus to the collection when focus moves into the cell's contents. Still digging

Reply:
So the reason why Virtualizer keeps moving focus back to the collection is because it thinks the cell isn't visible. This is because [getView](https://github.com/adobe/react-spectrum/blob/66d8eb4f4da3c073f4184f70ff6fcad47a1e8238/packages/%40react-aria/virtualizer/src/Virtualizer.tsx#L161) uses the layout's `getVisibleLayoutInfos` to check if the key is within view 
code path:
https://github.com/adobe/react-spectrum/blob/66d8eb4f4da3c073f4184f70ff6fcad47a1e8238/packages/%40react-stately/virtualizer/src/Virtualizer.ts#L644 -> https://github.com/adobe/react-spectrum/blob/66d8eb4f4da3c073f4184f70ff6fcad47a1e8238/packages/%40react-stately/virtualizer/src/Virtualizer.ts#L620 -> https://github.com/adobe/react-spectrum/blob/66d8eb4f4da3c073f4184f70ff6fcad47a1e8238/packages/%40react-stately/virtualizer/src/Virtualizer.ts#L625.
Since we are using ListLayout here, the only nodes it will check are the [rootNodes](https://github.com/adobe/react-spectrum/blob/66d8eb4f4da3c073f4184f70ff6fcad47a1e8238/packages/%40react-stately/layout/src/ListLayout.ts#L116) so when we attempt to move focus to the cell it thinks the cell is out of view

Not sure what I can do about this short of modifying ListLayout to track the child nodes as well... Is there anything I'm missing?